src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java

```diff
@@ -200,8 +202,13 @@ protected void updateCitizens()
         //Removes all citizens which already have a job.
         citizens = colony.getCitizens().values().stream()
                      .filter(citizen -> !citizen.isChild())
-                     .filter(citizen -> (citizen.getWorkBuilding() == null)
-                                          || building.getPosition().equals(citizen.getWorkBuilding())).sorted(Comparator.comparing(ICitizenDataView::getName))
+                     .filter(citizen -> {
+                         return (citizen.getWorkBuilding() == null)
+                                 || building.getPosition().equals(citizen.getWorkBuilding())
+                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("library")
+                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("archery")
+                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("combatacademy");
```

Comment:
Dunno if it's a huge deal here, but as a trick in the future, you can do "instanceof" on the Building for these type of checks and it's a lot less of a hassle than fighting with strings everywhere (especially for some of the cases where schematic names and class names are different).

Also lets developers do fun things with inheritance a bit simpler.

Reply:
This should be more generic. Create an interface like "IBuildingCanBeHiredFrom", add it to the views we want this in. And then just do this one check here.

Reply:
Can do.

Reply:
Alright. I'm obviously doing something wrong here. Because each building view for the three buildings in question is constructed by the AbstractBuildingWorker class, I extended that class into AbstractBuildingTraining, and made the view that AbstractBuildingTraining creates implement the IBuildingCanBeHiredFrom interface. Now, the code compiles and the check runs, but the instanceof isn't detecting the interface. I think. I'm going to go deeper into logging and debugging this, but figured I'd put this up there as a sheet anchor incase someone has an idea.

Reply:
You're in GUI world, so you probably do need to check against the BuildingWhatever.View.  See if : 

`|| colony.getBuilding(citizen.getWorkBuilding()) instanceof BuildingSchool.View 
|| colony.getBuilding(citizen.getWorkBuilding()) instanceof BuildingArchery.View
|| colony.getBuilding(citizen.getWorkBuilding()) instanceof BuildingCombatAcademy.View`

does what you need.

Reply:
I think I've got it. I thought they were implemented somewhere else, but they aren't. Thanks